packages/api-server/src/core/cache/Cache.ts

```diff
@@ -81,39 +81,34 @@ export default class Cache<T> {
 
       return cachedData;
     } else {
-      this.log.debug(`No hit cache of ${this.key}.`);
+      this.log.debug('No hit cache of %s.', this.key);
 
       if (this.onlyFromCache && !this.refreshCache) {
         throw new NeedPreFetchError(`Failed to get data, query ${this.key} can only be executed in advance.`)
       }
 
-      this.log.info(`Do query for key: ${this.key}.`);
+      this.log.info('Do query for key: %s.', this.key);
       return await this.fetchDataFromDB(fallback);
     }
   }
 
   private async fetchDataFromDB(fallback: () => Promise<CachedData<T>>):Promise<CachedData<T>> {
-    const result = await fallback()
-
-    // Write result to cache.
-
-    try {
-      await measure(cacheQueryTimer.labels({op: 'set'}), async () => {
-        if (this.cacheHours === -1) {
-          result.expiresAt = MAX_CACHE_TIME;
-          await this.cacheProvider.set(this.key, JSON.stringify(result));
-        } else {
-          result.expiresAt = result.finishedAt.plus({ hours: this.cacheHours })
-          await this.cacheProvider.set(this.key, JSON.stringify(result), {
-            EX: Math.round(this.cacheHours * 3600),
-          });
-        }
-      });
-      result.refresh = true;
-    } catch (err) {
-      this.log.error('Failed to write cache for key %s.', this.key, err)
-    }
-
+    const result = await fallback();
+    result.expiresAt = MAX_CACHE_TIME;
+
+    // Write result to cache async.
+    measure(cacheQueryTimer.labels({op: 'set'}), async () => {
```

Comment:
no `await` for this measurement?

Reply:
Maybe we should write the result to cache async to avoid the write latency of the cached table.

> The write latency of cached tables is high, because the cached table feature is implemented with a complex mechanism that requires a lease to be set for each cache. When there are multiple TiDB instances, one instance does not know whether the other instances have cached data. If an instance modifies the table data directly, the other instances read the old cache data. To ensure correctness, the cached table implementation uses a lease mechanism to ensure that the data is not modified before the lease expires. That is why the write latency is high.

Reply:
This will break runningCaches logic: if we don't wait writing cache, the next query with same cache key would start a new direct query to database in critical cases.

I prefer not to change this behavior until we have more tests on it.

Reply:
will fix in #956